pagegraph-cli/src/downstream_requests.rs

```diff
@@ -1,16 +1,29 @@
 //! Prints out all downstream network requests of a given edge from the graph.
 
 use pagegraph::{graph::{EdgeId, PageGraph}, types::EdgeType};
+use pagegraph::graph::DownstreamRequests;
+use pagegraph::types::NodeType;
 
 pub fn main(graph: &PageGraph, edge_id: EdgeId) {
-
-    let downstream_effects = graph.all_downstream_effects_of(graph.edges.get(&edge_id).unwrap()).into_iter().filter_map(|edge| {
-        if let EdgeType::RequestStart { request_id, .. } = edge.edge_type {
-            Some((request_id, format!("{}", edge.id)))
-        } else {
-            None
-        }
-    }).collect::<Vec<_>>();
-
-    println!("{}", serde_json::to_string(&downstream_effects).unwrap());
+    let edge = graph.edges.get(&edge_id).unwrap();
+    let all_downstream_requests = graph
+        .all_downstream_requests_nested(graph.edges.get(&edge_id).unwrap());
+    let node = graph.target_node(edge);
+    let url = match &node.node_type {
+        NodeType::Resource { url } => url,
+        _ => unreachable!()
+    };
+    match &edge.edge_type {
+        EdgeType::RequestStart {request_id, request_type, ..} => {
+            let top_level = DownstreamRequests {
+                request_id: request_id.clone(),
```

Comment:
nit: `request_id` is a `usize`, which implements `Copy`, so you don't need to call `clone()` here

Reply:
I'll dereference it then.